### PR TITLE
nvim: add AttachBuffer testcases

### PR DESCRIPTION
**Pull request by zchee**
Add `AttachBuffer` testcases.

### CODE REVIEW

**Comment by zchee** [comment]
ref: https://github.com/neovim/node-client/blob/c1660aa98bd18c689873f7b5c2693858af6bc1b2/packages/neovim/src/api/Buffer.test.ts#L413

**Comment by zchee** [comment]
@justinmk
This testcases are bit ugly. Will fix to more better test code.

**Comment by justinmk** [comment]
They seem fine to me, but ok :)

**Comment by zchee** [comment]
@justinmk
call `t.Fatal` into `goroutine` (In Python, threading? See https://tour.golang.org/concurrency) is ugly...
test is passed, but consider alter testcases :)

**Comment by zchee** [comment]
@justinmk changed called `t.Fatal` without goroutine. Ready to merge and PTAL.

Note: I'll send https://github.com/neovim/go-client/tree/api/9d77a07 PR later. then more adding testcase for new API and `AttachBuffer`.

File: nvim/nvim_test.go [hunk 1]
```diff
@@ -379,6 +381,115 @@ func TestAPI(t *testing.T) {
...
+						errc <- fmt.Errorf("bufLines = %+v, want %+v", bufLines, expected)
+					}
+					atomic.AddInt64(&numEvent, 1)
+				}
+			}
+		}()
+
+		go func() {
+			<-done
+			close(errc)
```
Comment (zchee) [inline nvim/nvim_test.go]: Note: catch the error on above goroutine.

File: nvim/nvim_test.go [hunk 2]
```diff
@@ -379,6 +381,115 @@ func TestAPI(t *testing.T) {
...
+		}()
+
+		for err := range errc {
+			if err != nil {
+				t.Fatal(err)
```
Comment (zchee) [inline nvim/nvim_test.go]: Note: catch the error on above goroutine.

**Comment by zchee** [comment]
@justinmk Sorry, fixed a bits.

**Comment by zchee** [comment]
@justinmk Travis CI is pending... That's why I said want to switch to CircleCI.. :(

**Comment by zchee** [comment]
@justinmk CI is passed. ready to merge!

**Comment by zchee** [comment]
@justinmk 
Thanks :)
I will send PR which fully supported Neovim latest API. pls review when if you have time.

---

Also, I considing to port gen_doc of neovim core and generate godoc comment automatically.
Current godoc document is here. There are some differences from the core C source document.
https://godoc.org/github.com/neovim/go-client/nvim#Nvim

But this repo is for Go, I will make some changes different to between the C and Go languages spec.

**Comment by justinmk** [comment]
> Also, I considing to port gen_doc of neovim core and generate godoc comment automatically.

Good idea. But, I don't think porting it is good. [nvim.net](https://github.com/neovim/nvim.net) did something similar. But that means we have a lot of code doing the same thing.

I think instead Nvim should return docs in the `nvIm_get_api_info` result. Although it should be optional somehow (disabled by default).

**Comment by zchee** [comment]
@justinmk
Ah, yes. port of gen_doc python script(means use doxygen and parses xml) to Go is one of the plans.
generates by `*.mpack`, use `nvim_get_api_info`, I was considering which one to use.

> Although it should be optional somehow (disabled by default).

Interesting, what arg(or whatever) enable output more details?

**Comment by justinmk** [comment]
> Interesting, what arg(or whatever) enable output more details?

It is not possible yet. But it's worth working on that rather than rewriting gen_vimdoc.py for all clients.

**Comment by zchee** [comment]
Ah I see. make sense.

**Pull request merged**